lib/ruby_lsp/requests/formatting.rb

```diff
@@ -60,10 +69,13 @@ def run
 
       sig { returns(T.nilable(String)) }
       def formatted_file
-        if defined?(Support::RuboCopFormattingRunner)
+        case @formatter
+        when "rubocop"
           Support::RuboCopFormattingRunner.instance.run(@uri, @document)
-        else
+        when "syntax_tree"
           SyntaxTree.format(@document.source)
+        else
```

Comment:
Looking at the VS code implementation `none` is also a valid response here. I think given the feature won't be enabled on `none` it won't matter, but wanted to comment just incase.

Reply:
Yeah, just avoid broadcasting the capability on `none`, which makes the editor not send any requests.